src/llmcompressor/transformers/sparsification/compressed_tensors_utils.py

```diff
@@ -14,19 +15,89 @@
+        if getattr(save_pretrained_method, "_overridden", False):
+            # `model.save_pretrained` has already been replaced, return.
+            return save_pretrained_method
+
+        # Keep a weak reference to the model class and unbound save_pretrained
+        # method so we can call the original
+        original_save_pretrained = save_pretrained_method.__func__
+        del save_pretrained_method
+
+        @wraps(original_save_pretrained)
+        def save_pretrained_wrapper(
+            save_directory: str,
+            **kwargs,
+        ):
+            """
+            Wrapper around PreTrainedModel.save_pretrained(), adds functionality for
+            saving models in a compressed format on disk. The compression format is
+            saved to the model's config file
+
+            :param save_directory: output directory to save model to
+            :param sparsity_config: optional sparsity config to compress model with,
+            if no config is provided it will be inferred from the model
+            :param quantization_format: optional compression format for quantized
+            models. If none is provided it will be inferred from the model
+            :param save_compressed: whether or not to compress the model on disk
+            :param skip_compression_stats: whether to skip the calculation of
+            compression statistics (such as global sparsity and sparsity structure) when
+            saving a model in dense format
+            :param kwargs: additional kwargs to pass on to model.save_pretrained
+            """
+            try:
+                trainer.save_model(output_dir=save_directory, _is_oneshot=True)
+            except AssertionError:
+                # fallback to this in the case of quantization
+                unwrap_and_export_model(
+                    model=trainer.model,
+                    accelerator=trainer.accelerator,
+                    output_dir=save_directory,
+                    tokenizer=tokenizer,
+                )
+                # only allow the main process move the state
+                # dicts to cpu
+                if trainer.accelerator.is_main_process:
+                    # assuming quantization is the last step
+                    # we no longer need the original model
+                    # and can safely delete it to save memory
+                    del trainer.model
+                    find_and_move_state_dicts_to_cpu(save_directory)
+
+        save_pretrained_wrapper._overriden = True
+        return save_pretrained_wrapper
 
-__all__ = ["modify_save_pretrained"]
+    # wrap save_pretrained
+    trainer.model.save_pretrained = save_pretrained_compressed(
+        trainer.model.save_pretrained
+    )
 
 
-def modify_save_pretrained(model: PreTrainedModel):
+def modify_save_pretrained(model: torch.nn.Module):
     """
     Overrides a PreTrainedModel's save_pretrained() method with a wrapped version that
```

Comment:
Misleading docstring!

Reply:
This is fine, it does wrap the save_pretrained

Reply:
The typing says model is an nn.Module, but the docstring says we override a PretrainedModel's save_pretrained?